modules/schedule/src/main/java/org/apache/ignite/internal/processors/schedule/ScheduleFutureImpl.java

```diff
@@ -90,7 +86,7 @@
     private int callCnt;
 
     /** De-schedule flag. */
-    private final AtomicBoolean descheduled = new AtomicBoolean(false);
+    private final AtomicBoolean descheduled = new AtomicBoolean(true);
```

Comment:
Could you explain why this flag is 'true' now?

Reply:
it was a bug, I believe. This flag reflects was the task actually scheduled by the underlying Scheduler or not. Initially task is not scheduled and it will be scheduled in org.apache.ignite.internal.processors.schedule.ScheduleFutureImpl#schedule0

And it can be delayed as we are supporting delayed scheduling (
org.apache.ignite.internal.processors.schedule.ScheduleFutureImpl#delay)

Now I've renamed flag to reverse name scheduled, it's a bit easier to understand I  suppose.